<?xml version="1.0"?>
<launch>

  <!-- send parameters to parameter server -->
  <rosparam command="load" ns="leg_detection" file="$(find cob_leg_detection)/ros/launch/leg_detection_params.yaml"/>

  <!-- leg detection front -->
  <node pkg="cob_leg_detection" ns="leg_detection_front" type="leg_detection" name="leg_detection_front" output="screen">	<!-- ns=namespace (arbitrary), type=name of executable, name=node name (arbitrary) -->
	<remap from="scan" to="/scan_front"/>
  </node>

  <!-- leg detection rear -->
  <node pkg="cob_leg_detection" ns="leg_detection_rear" type="leg_detection" name="leg_detection_rear" output="screen">	<!-- ns=namespace (arbitrary), type=name of executable, name=node name (arbitrary) -->
	<remap from="scan" to="/scan_rear"/>
  </node>

  <!-- leg detection top -->
  <node pkg="cob_leg_detection" ns="leg_detection_top" type="leg_detection" name="leg_detection_top" output="screen">	<!-- ns=namespace (arbitrary), type=name of executable, name=node name (arbitrary) -->
	<remap from="scan" to="/scan_top"/>
  </node>

  <!-- leg detection accumulator -->
  <node pkg="cob_leg_detection" ns="leg_detection" type="leg_detection_cob3" name="leg_detection_accumulator" output="screen">	<!-- ns=namespace (arbitrary), type=name of executable, name=node name (arbitrary) -->
	<remap from="detected_humans_laser_front" to="/leg_detection_front/detected_humans_laser"/>
	<remap from="detected_humans_laser_rear" to="/leg_detection_rear/detected_humans_laser"/>
	<remap from="detected_humans_laser_top" to="/leg_detection_top/detected_humans_laser"/>
  </node>


</launch>
